web/js/codeworld_shared.js

```diff
@@ -1203,4 +1203,19 @@ function printMessage(type, message) {
     }
 
     outputDiv.scrollTop = outputDiv.scrollHeight;
+
+    if (isRuntime && type === "error") {
+        parseCompileErrors(message).forEach(
+            (error) => {
+                runtimeErrors[error.fullText] = error;
+            }
+        )
+        let cm = window.codeworldEditor;
+        cm.performLint();
```

Comment:
Will this send another HTTP request to the server?  I don't think we want to do that.  Instead, I believe we want to cache the lint results for the current generation of code in a global variable.  The linter should check the generation against the latest one, and if it's the same, it will just work with the cached results.  Only if the source has been changed since then would we send a new request to the server.

Reply:
It is good remark. Will fix.

Reply:
I think all this is still irrelevant.  Again, instead of replacing the old errors, we'd like to just translate the warning to an error at build time, and then replace any deferred type error with the "giving up" message.  I'll resolve this.

Reply:
I almost done this.

Reply:
I about replacing on haskell side. Left one minor issue with correct matching "\[0-9]" pattern to replace group.